### ISSUE TITLE
Covariant return type causes false positive for method removal due to return type change.

### ISSUE THREAD

**Issue by busbey** [lvc/japi-compliance-checker#50]
Over in Apache HBase we had a change ([509c1b63](https://github.com/apache/hbase/commit/509c1b63ed8e051478e07f2047b9a09aec0b94ef)) that added a method to a supertype of a class such that the original class' method became an override via covariant return type.

In summary, before the change:
```
class Query {}

class Scan extends Query {
  public Scan setLoadColumnFamiliesOnDemand(boolean value);
}
```

and after the change:
```
class Query {
  public Query setLoadColumnFamiliesOnDemand(boolean value);
}

class Scan extends Query {
  public Scan setLoadColumnFamiliesOnDemand(boolean value);
}
```

Using japi-compliance-checker 2.4, this gets reported as a change in return type that will cause a High Severity binary compatibility problem for MethodNotFoundException and a Medium Severity source compatibility problem for downstream users.

e.g.

> <h2>Problems with Methods, High Severity <span class='failed'>&nbsp;1&nbsp;</span></h2><hr/>
>  <span class='jar'>hbase-client-1.2.6.jar</span>, <span class='cname'>Scan.class</span><br/>
>  <span class='pkg_t'>package</span> <span class='pkg'>org.apache.hadoop.hbase.client</span><br/>
>  <span class="section" onclick="sC(this, 'c_20')">
> <span class='ext'>[+]</span> Scan.setLoadColumnFamiliesOnDemand&#160;<span class='sym_p'><span>(&#160;boolean <span class='color_p'>value</span></span>&#160;)</span><span class='sym_p nowrap'> &#160;<b>:</b>&#160;&#160;Scan</span> <span class='failed'>&nbsp;1&nbsp;</span></span>
>  <br/>
>  <div id="c_20" style="display:none;">
>  <span class='mngl pleft'>org/apache/hadoop/hbase/client/Scan.setLoadColumnFamiliesOnDemand:(Z)Lorg/apache/hadoop/hbase/client/Scan;</span><br/>
>  <table class='ptable'><tr><th width='2%'></th><th width='47%'>Change</th><th>Effect</th></tr><tr>
>  <th>1</th>
>  <td>Return value type has been changed from <b>Scan</b> to <b>Query</b>.</td>
>  <td>This method has been removed because the return type is part of the method signature. A client program may be interrupted by <b>NoSuchMethodError</b> exception.</td>
>  </tr>
> </table><br/></div>
 
Examining the class file for the updated Scan class shows that since the method now has a covariant return type two method signatures are included:

```
busbey$ javap -cp target/compat-check/dst/hbase-client/target/hbase-client-1.2.7.jar org.apache.hadoop.hbase.client.Scan | grep setLoadColumnFamiliesOnDemand
  public org.apache.hadoop.hbase.client.Scan setLoadColumnFamiliesOnDemand(boolean);
  public org.apache.hadoop.hbase.client.Query setLoadColumnFamiliesOnDemand(boolean);
```

japi-compliance-checker appears to mistakenly just pick up the second method signature and thus the compatibility issues.
